### ISSUE TITLE
golint compliance

### ISSUE THREAD

**Issue by VadimKulagin** [deepmap/oapi-codegen#58]
The generated code and the code of the project itself causes a lot of warnings in the GoLand IDE and when checking with [golint](https://github.com/golang/lint).

How about bringing code into line with golint?

I could help with that.

```
golint ./...
examples/petstore-expanded/chi/petstore_test.go:16:6: func DoJson should be DoJSON
examples/petstore-expanded/chi/api/petstore.go:12:6: exported type PetStore should have comment or be unexported
examples/petstore-expanded/chi/api/petstore.go:14:2: struct field NextId should be NextID
examples/petstore-expanded/chi/api/petstore.go:18:1: exported function NewPetStore should have comment or be unexported
examples/petstore-expanded/chi/api/petstore.go:36:1: comment on exported method PetStore.FindPets should be of the form "FindPets ..."
examples/petstore-expanded/chi/api/petstore.go:71:1: exported method PetStore.AddPet should have comment or be unexported
examples/petstore-expanded/chi/api/petstore.go:100:1: exported method PetStore.FindPetById should have comment or be unexported
examples/petstore-expanded/chi/api/petstore.go:100:20: method FindPetById should be FindPetByID
examples/petstore-expanded/chi/api/petstore.go:116:1: exported method PetStore.DeletePet should have comment or be unexported
examples/petstore-expanded/echo/petstore_test.go:80:2: var petId should be petID
examples/petstore-expanded/echo/petstore_test.go:110:2: var petId2 should be petID2
examples/petstore-expanded/echo/api/petstore.go:28:6: exported type PetStore should have comment or be unexported
examples/petstore-expanded/echo/api/petstore.go:30:2: struct field NextId should be NextID
examples/petstore-expanded/echo/api/petstore.go:34:1: exported function NewPetStore should have comment or be unexported
examples/petstore-expanded/echo/api/petstore.go:52:1: comment on exported method PetStore.FindPets should be of the form "FindPets ..."
examples/petstore-expanded/echo/api/petstore.go:83:1: exported method PetStore.AddPet should have comment or be unexported
examples/petstore-expanded/echo/api/petstore.go:122:1: exported method PetStore.FindPetById should have comment or be unexported
examples/petstore-expanded/echo/api/petstore.go:122:20: method FindPetById should be FindPetByID
examples/petstore-expanded/echo/api/petstore.go:122:50: method parameter petId should be petID
examples/petstore-expanded/echo/api/petstore.go:134:1: exported method PetStore.DeletePet should have comment or be unexported
examples/petstore-expanded/internal/doc.go:1:1: package comment should be of the form "Package internal ..."
internal/test/components/components_test.go:10:6: func assertJsonEqual should be assertJSONEqual
internal/test/issues/issue-52/doc.go:1:1: don't use an underscore in package name
internal/test/issues/issue-52/issue_test.go:1:1: don't use an underscore in package name
pkg/codegen/codegen.go:242:1: exported function GenerateTypeDefinitions should have comment or be unexported
pkg/codegen/codegen.go:285:1: comment on exported function GenerateTypesForSchemas should be of the form "GenerateTypesForSchemas ..."
pkg/codegen/codegen.go:309:1: comment on exported function GenerateTypesForParameters should be of the form "GenerateTypesForParameters ..."
pkg/codegen/codegen.go:341:1: comment on exported function GenerateTypesForResponses should be of the form "GenerateTypesForResponses ..."
pkg/codegen/codegen.go:380:1: comment on exported function GenerateTypesForRequestBodies should be of the form "GenerateTypesForRequestBodies ..."
pkg/codegen/codegen.go:418:1: comment on exported function GenerateTypes should be of the form "GenerateTypes ..."
pkg/codegen/codegen.go:441:1: comment on exported function GenerateImports should be of the form "GenerateImports ..."
pkg/codegen/codegen.go:465:1: comment on exported function GenerateAdditionalPropertyBoilerplate should be of the form "GenerateAdditionalPropertyBoilerplate ..."
pkg/codegen/inline.go:1:1: package comment should be of the form "Package codegen ..."
pkg/codegen/inline.go:27:1: comment on exported function GenerateInlinedSpec should be of the form "GenerateInlinedSpec ..."
pkg/codegen/operations.go:1:1: package comment should be of the form "Package codegen ..."
pkg/codegen/operations.go:28:6: exported type ParameterDefinition should have comment or be unexported
pkg/codegen/operations.go:36:1: comment on exported method ParameterDefinition.TypeDef should be of the form "TypeDef ..."
pkg/codegen/operations.go:44:1: comment on exported method ParameterDefinition.JsonTag should be of the form "JsonTag ..."
pkg/codegen/operations.go:47:32: method JsonTag should be JSONTag
pkg/codegen/operations.go:50:9: if block ends with a return statement, so drop this else and outdent its block
pkg/codegen/operations.go:55:1: exported method ParameterDefinition.IsJson should have comment or be unexported
pkg/codegen/operations.go:55:32: method IsJson should be IsJSON
pkg/codegen/operations.go:64:1: exported method ParameterDefinition.IsPassThrough should have comment or be unexported
pkg/codegen/operations.go:75:1: exported method ParameterDefinition.IsStyled should have comment or be unexported
pkg/codegen/operations.go:80:1: exported method ParameterDefinition.Style should have comment or be unexported
pkg/codegen/operations.go:96:1: exported method ParameterDefinition.Explode should have comment or be unexported
pkg/codegen/operations.go:111:1: exported method ParameterDefinition.GoVariableName should have comment or be unexported
pkg/codegen/operations.go:122:1: exported method ParameterDefinition.GoName should have comment or be unexported
pkg/codegen/operations.go:126:1: exported method ParameterDefinition.IndirectOptional should have comment or be unexported
pkg/codegen/operations.go:130:6: exported type ParameterDefinitions should have comment or be unexported
pkg/codegen/operations.go:132:1: exported method ParameterDefinitions.FindByName should have comment or be unexported
pkg/codegen/operations.go:141:1: comment on exported function DescribeParameters should be of the form "DescribeParameters ..."
pkg/codegen/operations.go:179:6: exported type SecurityDefinition should have comment or be unexported
pkg/codegen/operations.go:184:1: exported function DescribeSecurityDefinition should have comment or be unexported
pkg/codegen/operations.go:196:1: comment on exported type OperationDefinition should be of the form "OperationDefinition ..." (with optional leading article)
pkg/codegen/operations.go:198:2: struct field OperationId should be OperationID
pkg/codegen/operations.go:214:1: comment on exported method OperationDefinition.Params should be of the form "Params ..."
pkg/codegen/operations.go:222:1: comment on exported method OperationDefinition.AllParams should be of the form "AllParams ..."
pkg/codegen/operations.go:230:1: comment on exported method OperationDefinition.RequiresParamObject should be of the form "RequiresParamObject ..."
pkg/codegen/operations.go:237:1: comment on exported method OperationDefinition.HasBody should be of the form "HasBody ..."
pkg/codegen/operations.go:244:1: comment on exported method OperationDefinition.SummaryAsComment should be of the form "SummaryAsComment ..."
pkg/codegen/operations.go:257:1: comment on exported method OperationDefinition.GetResponseTypeDefinitions should be of the form "GetResponseTypeDefinitions ..."
pkg/codegen/operations.go:315:1: comment on exported type RequestBodyDefinition should be of the form "RequestBodyDefinition ..." (with optional leading article)
pkg/codegen/operations.go:335:1: comment on exported method RequestBodyDefinition.TypeDef should be of the form "TypeDef ..."
pkg/codegen/operations.go:340:1: comment on exported method RequestBodyDefinition.CustomType should be of the form "CustomType ..."
pkg/codegen/operations.go:347:1: comment on exported method RequestBodyDefinition.Suffix should be of the form "Suffix ..."
pkg/codegen/operations.go:358:1: comment on exported function FilterParameterDefinitionByType should be of the form "FilterParameterDefinitionByType ..."
pkg/codegen/operations.go:451:1: comment on exported function GenerateBodyDefinitions should be of the form "GenerateBodyDefinitions ..."
pkg/codegen/operations.go:515:1: exported function GenerateTypeDefsForOperation should have comment or be unexported
pkg/codegen/operations.go:533:1: comment on exported function GenerateParamsTypes should be of the form "GenerateParamsTypes ..."
pkg/codegen/operations.go:572:1: comment on exported function GenerateTypesForOperations should be of the form "GenerateTypesForOperations ..."
pkg/codegen/operations.go:665:1: comment on exported function GenerateServerInterface should be of the form "GenerateServerInterface ..."
pkg/codegen/operations.go:682:1: comment on exported function GenerateWrappers should be of the form "GenerateWrappers ..."
pkg/codegen/operations.go:701:1: comment on exported function GenerateRegistration should be of the form "GenerateRegistration ..."
pkg/codegen/operations.go:719:1: comment on exported function GenerateClient should be of the form "GenerateClient ..."
pkg/codegen/operations.go:737:1: comment on exported function GenerateClientWithResponses should be of the form "GenerateClientWithResponses ..."
pkg/codegen/schema.go:11:1: comment on exported type Schema should be of the form "Schema ..." (with optional leading article)
pkg/codegen/schema.go:24:1: exported method Schema.IsRef should have comment or be unexported
pkg/codegen/schema.go:28:1: exported method Schema.TypeDecl should have comment or be unexported
pkg/codegen/schema.go:35:1: exported method Schema.MergeProperty should have comment or be unexported
pkg/codegen/schema.go:46:1: exported method Schema.GetAdditionalTypeDefs should have comment or be unexported
pkg/codegen/schema.go:55:6: exported type Property should have comment or be unexported
pkg/codegen/schema.go:56:2: struct field JsonFieldName should be JSONFieldName
pkg/codegen/schema.go:61:1: exported method Property.GoFieldName should have comment or be unexported
pkg/codegen/schema.go:65:1: exported method Property.GoTypeDef should have comment or be unexported
pkg/codegen/schema.go:73:6: exported type TypeDefinition should have comment or be unexported
pkg/codegen/schema.go:75:2: struct field JsonName should be JSONName
pkg/codegen/schema.go:80:1: exported function PropertiesEqual should have comment or be unexported
pkg/codegen/schema.go:84:1: exported function GenerateGoSchema should have comment or be unexported
pkg/codegen/schema.go:200:9: if block ends with a return statement, so drop this else and outdent its block
pkg/codegen/schema.go:256:1: comment on exported type SchemaDescriptor should be of the form "SchemaDescriptor ..." (with optional leading article)
pkg/codegen/schema.go:263:6: exported type FieldDescriptor should have comment or be unexported
pkg/codegen/schema.go:267:2: struct field JsonName should be JSONName
pkg/codegen/schema.go:271:1: comment on exported function GenFieldsFromProperties should be of the form "GenFieldsFromProperties ..."
pkg/codegen/schema.go:287:1: exported function GenStructFromSchema should have comment or be unexported
pkg/codegen/schema.go:306:1: comment on exported function MergeSchemas should be of the form "MergeSchemas ..."
pkg/codegen/schema.go:359:1: comment on exported function GenStructFromAllOf should be of the form "GenStructFromAllOf ..."
pkg/codegen/template_helpers.go:1:1: package comment should be of the form "Package codegen ..."
pkg/codegen/template_helpers.go:222:1: comment on exported var TemplateFunctions should be of the form "TemplateFunctions ..."
pkg/codegen/utils.go:1:1: package comment should be of the form "Package codegen ..."
pkg/codegen/utils.go:42:1: comment on exported function UppercaseFirstCharacter should be of the form "UppercaseFirstCharacter ..."
pkg/codegen/utils.go:53:1: comment on exported function LowercaseFirstCharacter should be of the form "LowercaseFirstCharacter ..."
pkg/codegen/utils.go:63:1: comment on exported function ToCamelCase1 should be of the form "ToCamelCase1 ..."
pkg/codegen/utils.go:89:1: comment on exported function ToCamelCase should be of the form "ToCamelCase ..."
pkg/codegen/utils.go:104:1: comment on exported function SortedSchemaKeys should be of the form "SortedSchemaKeys ..."
pkg/codegen/utils.go:117:1: comment on exported function SortedPathsKeys should be of the form "SortedPathsKeys ..."
pkg/codegen/utils.go:130:1: comment on exported function SortedOperationsKeys should be of the form "SortedOperationsKeys ..."
pkg/codegen/utils.go:142:1: comment on exported function SortedResponsesKeys should be of the form "SortedResponsesKeys ..."
pkg/codegen/utils.go:154:1: comment on exported function SortedContentKeys should be of the form "SortedContentKeys ..."
pkg/codegen/utils.go:166:1: comment on exported function SortedStringKeys should be of the form "SortedStringKeys ..."
pkg/codegen/utils.go:178:1: comment on exported function SortedParameterKeys should be of the form "SortedParameterKeys ..."
pkg/codegen/utils.go:190:1: exported function SortedRequestBodyKeys should have comment or be unexported
pkg/codegen/utils.go:201:1: comment on exported function StringInArray should be of the form "StringInArray ..."
pkg/codegen/utils.go:212:1: comment on exported function RefPathToGoType should be of the form "RefPathToGoType ..."
pkg/codegen/utils.go:232:1: comment on exported function SwaggerUriToEchoUri should be of the form "SwaggerUriToEchoUri ..."
pkg/codegen/utils.go:243:6: func SwaggerUriToEchoUri should be SwaggerURIToEchoURI
pkg/codegen/utils.go:247:1: comment on exported function SwaggerUriToChiUri should be of the form "SwaggerUriToChiUri ..."
pkg/codegen/utils.go:258:6: func SwaggerUriToChiUri should be SwaggerURIToChiURI
pkg/codegen/utils.go:262:1: comment on exported function OrderedParamsFromUri should be of the form "OrderedParamsFromUri ..."
pkg/codegen/utils.go:264:6: func OrderedParamsFromUri should be OrderedParamsFromURI
pkg/codegen/utils.go:273:1: comment on exported function ReplacePathParamsWithStr should be of the form "ReplacePathParamsWithStr ..."
pkg/codegen/utils.go:278:1: comment on exported function SortParamsByPath should be of the form "SortParamsByPath ..."
pkg/codegen/utils.go:298:1: comment on exported function IsGoKeyword should be of the form "IsGoKeyword ..."
pkg/codegen/utils.go:336:1: comment on exported function SchemaNameToTypeName should be of the form "SchemaNameToTypeName ..."
pkg/codegen/utils.go:347:1: comment on exported function SchemaHasAdditionalProperties should be of the form "SchemaHasAdditionalProperties ..."
pkg/codegen/utils.go:363:1: comment on exported function PathToTypeName should be of the form "PathToTypeName ..."
pkg/middleware/oapi_validate.go:29:7: exported const EchoContextKey should have comment or be unexported
pkg/middleware/oapi_validate.go:30:7: exported const UserDataKey should have comment or be unexported
pkg/middleware/oapi_validate.go:36:1: comment on exported function OapiValidatorFromYamlFile should be of the form "OapiValidatorFromYamlFile ..."
pkg/middleware/oapi_validate.go:51:1: comment on exported function OapiRequestValidator should be of the form "OapiRequestValidator ..."
pkg/middleware/oapi_validate.go:64:1: comment on exported function OapiRequestValidatorWithOptions should be of the form "OapiRequestValidatorWithOptions ..."
pkg/middleware/oapi_validate.go:78:1: comment on exported function ValidateRequestFromContext should be of the form "ValidateRequestFromContext ..."
pkg/middleware/oapi_validate.go:107:20: should not use basic type string as key in context.WithValue
pkg/middleware/oapi_validate.go:112:20: should not use basic type string as key in context.WithValue
pkg/middleware/oapi_validate.go:136:1: comment on exported function GetEchoContext should be of the form "GetEchoContext ..."
pkg/middleware/oapi_validate.go:150:1: exported function GetUserData should have comment or be unexported
pkg/runtime/bindparam.go:1:1: package comment should be of the form "Package runtime ..."
pkg/runtime/bindparam.go:27:1: comment on exported function BindStyledParameter should be of the form "BindStyledParameter ..."
pkg/runtime/bindparam.go:105:10: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:134:10: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:154:10: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:234:1: comment on exported function BindQueryParameter should be of the form "BindQueryParameter ..."
pkg/runtime/bindparam.go:310:13: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:328:13: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:347:10: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:353:12: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindparam.go:374:12: if block ends with a return statement, so drop this else and outdent its block
pkg/runtime/bindstring.go:1:1: package comment should be of the form "Package runtime ..."
pkg/runtime/bindstring.go:24:1: comment on exported function BindStringToObject should be of the form "BindStringToObject ..."
pkg/runtime/router.go:1:1: package comment should be of the form "Package runtime ..."
pkg/runtime/router.go:20:1: comment on exported type EchoRouter should be of the form "EchoRouter ..." (with optional leading article)
pkg/runtime/styleparam.go:1:1: package comment should be of the form "Package runtime ..."
pkg/runtime/styleparam.go:25:1: comment on exported function StyleParam should be of the form "StyleParam ..."
pkg/testutil/request_helpers.go:1:1: package comment should be of the form "Package testutil ..."
pkg/testutil/request_helpers.go:39:1: exported function NewRequest should have comment or be unexported
pkg/testutil/request_helpers.go:45:1: comment on exported type RequestBuilder should be of the form "RequestBuilder ..." (with optional leading article)
pkg/testutil/request_helpers.go:55:1: comment on exported method RequestBuilder.WithMethod should be of the form "WithMethod ..."
pkg/testutil/request_helpers.go:62:1: exported method RequestBuilder.Get should have comment or be unexported
pkg/testutil/request_helpers.go:66:1: exported method RequestBuilder.Post should have comment or be unexported
pkg/testutil/request_helpers.go:70:1: exported method RequestBuilder.Put should have comment or be unexported
pkg/testutil/request_helpers.go:74:1: exported method RequestBuilder.Delete should have comment or be unexported
pkg/testutil/request_helpers.go:78:1: comment on exported method RequestBuilder.WithHeader should be of the form "WithHeader ..."
pkg/testutil/request_helpers.go:84:1: exported method RequestBuilder.WithContentType should have comment or be unexported
pkg/testutil/request_helpers.go:88:1: exported method RequestBuilder.WithJsonContentType should have comment or be unexported
pkg/testutil/request_helpers.go:88:26: method WithJsonContentType should be WithJSONContentType
pkg/testutil/request_helpers.go:92:1: exported method RequestBuilder.WithAccept should have comment or be unexported
pkg/testutil/request_helpers.go:96:1: exported method RequestBuilder.WithAcceptJson should have comment or be unexported
pkg/testutil/request_helpers.go:96:26: method WithAcceptJson should be WithAcceptJSON
pkg/testutil/request_helpers.go:102:1: exported method RequestBuilder.WithBody should have comment or be unexported
pkg/testutil/request_helpers.go:107:1: comment on exported method RequestBuilder.WithJsonBody should be of the form "WithJsonBody ..."
pkg/testutil/request_helpers.go:109:26: method WithJsonBody should be WithJSONBody
pkg/testutil/request_helpers.go:118:1: comment on exported method RequestBuilder.WithCookie should be of the form "WithCookie ..."
pkg/testutil/request_helpers.go:124:1: exported method RequestBuilder.WithCookieNameValue should have comment or be unexported
pkg/testutil/request_helpers.go:128:1: comment on exported method RequestBuilder.Go should be of the form "Go ..."
pkg/testutil/request_helpers.go:157:1: comment on exported type CompletedRequest should be of the form "CompletedRequest ..." (with optional leading article)
pkg/testutil/request_helpers.go:163:1: comment on exported method CompletedRequest.UnmarshalBodyToObject should be of the form "UnmarshalBodyToObject ..."
pkg/testutil/request_helpers.go:179:1: comment on exported method CompletedRequest.UnmarshalJsonToObject should be of the form "UnmarshalJsonToObject ..."
pkg/testutil/request_helpers.go:181:28: method UnmarshalJsonToObject should be UnmarshalJSONToObject
pkg/testutil/request_helpers.go:185:1: comment on exported method CompletedRequest.Code should be of the form "Code ..."
```

**Comment by deepmap-marcinr** [comment]
I wouldn't worry about fixing this in generated code, we simply can't predict it well enough yet for all possible api specs, but we could clean up lint warnings.

It looks like those are mostly issues with comments - nothing dangerous, so I don't really care :) I've been focusing on having stuff actually functional, which is hard enough with this project.

**Comment by carmo-evan** [comment]
This is quite annoying specially when integrating the generated code with a pre-existing codebase that does abide by the `golint` rules. In my case, I'm running into a lot of annoying scenarios with acronyms not being capitalized. It shouldn't be terrible to do a pass of `golint` plus some sort of autofix tool before generating the final output.

**Comment by deepmap-marcinr** [comment]
Yeah, we can add some optional flag to fix this. It would be easier to do at generation time than a post-process, but we'd have to make it optional and default-off, since this would break lots of existing code.

**Comment by carmo-evan** [comment]
I could take a stab at it if this is not on top of your priority list.

**Comment by jamietanna** [comment]
Looking at this for `staticcheck`, it's a fairly limited list of things to fix, so I think this should be doable 😄 

```
staticcheck ./examples/...
examples/authenticated-api/echo/server/jwt_authenticator.go:20:23: error strings should not be capitalized (ST1005)
examples/authenticated-api/echo/server/jwt_authenticator.go:21:28: error strings should not be capitalized (ST1005)
examples/authenticated-api/echo/server/jwt_authenticator.go:22:24: error strings should not be capitalized (ST1005)
examples/petstore-expanded/gin/api/petstore.go:108:2: redundant return statement (S1023)
examples/petstore-expanded/strict/api/petstore-server.gen.go:427:45: error strings should not be capitalized (ST1005)
examples/petstore-expanded/strict/api/petstore-server.gen.go:464:45: error strings should not be capitalized (ST1005)
examples/petstore-expanded/strict/api/petstore-server.gen.go:494:45: error strings should not be capitalized (ST1005)
examples/petstore-expanded/strict/api/petstore-server.gen.go:526:45: error strings should not be capitalized (ST1005)
```

**Comment by jamietanna** [comment]
See work started in #722

**Comment by jamietanna** [comment]
Closing as this should now be resolved by work in #722 and some other PRs!
